aldryn_newsblog/views.py

```diff
@@ -71,8 +77,28 @@ def get_queryset(self):
         return qs
 
 
+class AppHookCheckMixin(object):
+
+    def dispatch(self, request, *args, **kwargs):
+        language = translation.get_language_from_request(
+            request, check_path=True)
+        site_id = getattr(get_current_site(request), 'id', None)
+        self.valid_languages = get_valid_languages(
+            self.namespace,
+            language_code=language,
+            site_id=site_id)
+        return super(AppHookCheckMixin, self).dispatch(
+            request, *args, **kwargs)
+
+    def get_queryset(self):
+        # filter available objects to contain only resolvable for current
+        # language
+        qs = super(AppHookCheckMixin, self).get_queryset()
+        return qs.translated(*self.valid_languages)
```

Comment:
just a small note that in parler .translated() returns a queryset that can't be combined with other filters that access translated fields.